zk/stages/stage_sequence_execute_utils.go

```diff
@@ -489,3 +508,78 @@ func (bdc *BlockDataChecker) AddTransactionData(txL2Data []byte) bool {
 
 	return false
 }
+
+type txMatadata struct {
+	blockNum int
+	txIndex  int
+}
+
+type ResequenceBatchJob struct {
+	batchToProcess  []*dsTypes.FullL2Block
+	StartBlockIndex int
+	StartTxIndex    int
+	txIndexMap      map[common.Hash]txMatadata
+}
+
+func NewResequenceBatchJob(batch []*dsTypes.FullL2Block) *ResequenceBatchJob {
+	return &ResequenceBatchJob{
+		batchToProcess:  batch,
```

Comment:
maybe a misleading name?  blocksToProcess perhaps?  Similar for the param name being batch.  Typically this means a batch number rather than a collection of blocks.

Reply:
This struct represent the original batch that might be preserved or split after resequencing. The name seems to match to what it contains.